detox/index.d.ts

```diff
@@ -287,6 +334,35 @@ declare global {
              * The value will be undefined until the device is properly prepared (i.e. in detox.init()).
              */
             name: string;
+
+            /**
+             * Select the current app (relevant only to multi-app configs) by its name.
+             * After execution, all app-specific device methods will target the selected app.
+             *
+             * @see DetoxAppConfig#name
+             * @example
+             * await device.selectApp('passenger');
+             * await device.launchApp(); // passenger
+             * // ... run tests for the passenger app
+             * await device.uninstallApp(); // passenger
+             * await device.selectApp('driver');
+             * await device.installApp(); // driver
+             * await device.launchApp(); // driver
+             * // ... run tests for the driver app
+             * await device.terminateApp(); // driver
+             */
+            selectApp(app: string): Promise<void>;
+
+            /**
+             * @protected
+             * Unselect the current app.
+             * After execution, all app-specific device methods will be throwing an error.
+             *
+             * @example
+             * await device.selectApp(null); // explicitly unselects any app
+             */
+             selectApp(app: null): Promise<void>;
```

Comment:
I'm kind-of picking my brain trying to think whether this is really needed, because none of the API's are really app-specific (or are they?)

Reply:
This is a protected method - I use it internally with multiple apps config to unselect the default app so you always will be getting an error unless you explicitly select your app after Detox init.

Do you think it is better to remove it at all? I don't have a strong opinion here, so pls let me know.

Reply:
Oh, it _is_ `@protected` indeed :)
Your call, then!

But to be sure I understand correctly: eventually, this is so as to protect users from forgetting to select an app when multiple apps have been configured in the selected configuration?

Reply:
@d4vidi, exactly for that purpose — we can't decide what application should be default there.